pkg/apis/kudo/v1beta1/instance_types_helpers.go

```diff
@@ -78,6 +80,12 @@ func (i *Instance) UpdateInstanceStatus(ps *PlanStatus, updatedTimestamp *metav1
 			i.Spec.PlanExecution.Status = ps.Status
 		}
 	}
+	if i.Spec.PlanExecution.Status.IsFinished() {
+		i.SetReadinessTrue("")
+	}
+	if i.Spec.PlanExecution.Status == ExecutionFatalError {
+		i.SetReadinessFatalError("")
```

Comment:
IMO that's too much, because the information is already in the plan status. The message is usually set from the actual resource health messages, i.e. `status.IsReady` messages. Plan status information isn't individual resource health.